indexer/proc_stableToken.go

```diff
@@ -0,0 +1,125 @@
+package indexer
+
+import (
+	"context"
+
+	"github.com/celo-org/celo-blockchain/accounts/abi/bind"
+	"github.com/celo-org/celo-blockchain/log"
+	"github.com/celo-org/eksportisto/metrics"
+	"github.com/celo-org/eksportisto/utils"
+	"github.com/celo-org/kliento/celotokens"
+	"github.com/celo-org/kliento/contracts"
+	"github.com/celo-org/kliento/contracts/helpers"
+	"github.com/celo-org/kliento/registry"
+	"github.com/go-errors/errors"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type stableTokenProcessorFactory struct{}
+
+func (stableTokenProcessorFactory) InitProcessors(
+	ctx context.Context,
+	handler *blockHandler,
+) ([]Processor, error) {
+	processors := make([]Processor, 0, 20)
+
+	celoTokenContracts, err := handler.celoTokens.GetContracts(ctx, handler.blockNumber, true)
+	if err != nil {
+		return nil, errors.Wrap(err, 1)
+	}
+
+
+	for stableToken, celoTokenContract := range celoTokenContracts {
+		// If a token's contract has not been registered with the Registry
+		// yet, the contract will be nil. Ignore this.
+		if celoTokenContract == nil {
+			continue
+		}
+		stableTokenRegistryID, err := celotokens.GetRegistryID(stableToken)
+		if err != nil {
+			return nil, errors.Wrap(err, 1)
+		}
+
+		stableTokenContract, err := handler.registry.GetContractByID(ctx, string(stableTokenRegistryID), handler.blockNumber)
+		if err != nil {
+			return nil, errors.Wrap(err, 1)
+		}
+
+		var stableContractVariable = stableTokenContract.(*contracts.StableToken)
+
+		totalSupplyGauge, err := metrics.CeloTokenSupply.GetMetricWithLabelValues(string(stableToken))
+		if err != nil {
+			return nil, errors.Wrap(err, 1)
+		}
+
+		processors = append(processors, &stableTokenProcessor{
+			blockHandler:     handler,
+			logger:           handler.logger.New("processor", "stableToken", "contract", string(stableTokenRegistryID)),
+			stableTokenContract:    stableContractVariable,
+			stableTokenRegistryID:  stableTokenRegistryID,
+			totalSupplyGauge: totalSupplyGauge,
+		})
+	}
+	return processors, nil
+}
+
+type stableTokenProcessor struct {
+	*blockHandler
+	logger           log.Logger
+	stableTokenContract    *contracts.StableToken
+	stableTokenRegistryID  registry.ContractID
+	totalSupplyGauge prometheus.Gauge
```

Comment:
Nit/style: I think in other files we're keeping these aligned. Would you mind trying to run `go fmt`?